### ISSUE TITLE
SD-card code hardcodes SPI GPIO pins (AUD-3001)

### ISSUE THREAD

**Issue by ropg** [espressif/esp-adf#617]
From `esp-adf/components/esp_peripherals/lib/sdcard/sdcard.c`:

```c
#define PIN_NUM_MISO 2
#define PIN_NUM_MOSI 15
#define PIN_NUM_CLK  14
#define PIN_NUM_CS   13
```

See [here](https://github.com/ropg/m5core2_adf#sd-card-spi-pins) for more information.

**Comment by HengYongChao** [comment]
Hi @ropg 

I have to admit that this code is not good. Thanks for your advice!

**Comment by and-elf** [comment]
A simple fix would be too use get_spi_pins, wouldn't it? I guess most use this for a generic SPI (boy the same cs pin as SD card).. maybe a generic get_sd_pins would be nice?

**Comment by jason-mao** [comment]
Fixed it on 80fd0163d9bc7613fabed42d4fa782485b3a0b44
